test/integration/containerized/test_cli_containerized.py

```diff
@@ -29,3 +33,40 @@ def test_playbook_run(cli, skip_if_no_podman):
 def test_provide_env_var(cli, skip_if_no_podman, test_data_dir):
     r = cli(['run', os.path.join(test_data_dir, 'job_env'), '-p', 'printenv.yml'])
     assert 'gifmyvqok2' in r.stdout, r.stdout
+
+
+@pytest.mark.serial
+def test_cli_kill_cleanup(cli, test_data_dir, container_runtime_installed):
+    unique_string = str(uuid4()).replace('-', '')
+    ident = f'kill_test_{unique_string}'
+    pdd = os.path.join(test_data_dir, 'sleep')
+    cli_args = ['start', pdd, '-p', 'sleep.yml', '--ident', ident,
+                '--process-isolation', '--process-isolation-executable', container_runtime_installed]
+    cli(cli_args)
+
+    def container_is_running():
+        r = cli([container_runtime_installed, 'ps', '-f', f'name=ansible_runner_{ident}', '--format={{.Names}}'], bare=True)
+        return ident in r.stdout
+
+    tries = 5
+    for i in range(tries):
+        if container_is_running():
+            break
+        time.sleep(1)
+    else:
+        assert container_is_running()
+
+    # give playbook execution time to start
+    time.sleep(5)
```

Comment:
Is there a better way to detect this other than artificial sleeps? Intentionally slowing down tests is always something to avoided if possible.

Reply:
If being able to reproduce this with in integration test is unreliable, then unit and functional tests would be more beneficial until we have a better understanding of the exact failure scenario.

Reply:
Deleted (and thanks for the speculation about what might account for the variation in reproducibility between environments).

Reply:
✅